src/languages.ts

```diff
@@ -0,0 +1,128 @@
+export const languages = [
+  "abap",
+  "aes",
+  "apex",
+  "azcli",
+  "bat",
+  "bicep",
+  "c",
+  "cameligo",
+  "clojure",
+  "coffeescript",
+  "cpp",
+  "csharp",
+  "csp",
+  "css",
+  "dart",
+  "dockerfile",
+  "ecl",
+  "elixir",
+  "flow9",
+  "fsharp",
+  "go",
+  "graphql",
+  "handlebars",
+  "hcl",
+  "html",
+  "ini",
+  "java",
+  "javascript",
+  "json",
+  "julia",
+  "kotlin",
+  "less",
+  "lexon",
+  "liquid",
+  "lua",
+  "m3",
+  "markdown",
+  "mips",
+  "msdax",
+  "mysql",
+  "objective-c",
+  "pascal",
+  "pascaligo",
+  "perl",
+  "pgsql",
+  "php",
+  "pla",
+  "plaintext",
+  "postiats",
+  "powerquery",
+  "powershell",
+  "proto",
+  "pug",
+  "python",
+  "qsharp",
+  "r",
+  "razor",
+  "redis",
+  "redshift",
+  "restructuredtext",
+  "ruby",
+  "rust",
+  "sb",
+  "scala",
+  "scheme",
+  "scss",
+  "shell",
+  "sol",
+  "sparql",
+  "sql",
+  "st",
+  "swift",
+  "systemverilog",
+  "tcl",
+  "twig",
+  "typescript",
+  "vb",
+  "verilog",
+  "xml",
+  "yaml"
+] as const;
+
+export type Language = typeof languages[number];
+
+/** The default extension of the known programming languages (to be used on download) */
+const defaultExtensions: {[key in Language]?: string} = {
```

Comment:
We could either remove the question mark here, or actually generate `languages` from this (Object.keys()), both would have the result that every language has at least a default extension to be used on download and to autodetect the language

Reply:
@ekzhang ping here. Generating the languages from the extensions might be helpful.